lib/spack/spack/installer.py

```diff
@@ -2311,8 +2322,8 @@ def cache_only(self):
 
     @property
     def key(self):
-        """The key is the tuple (# uninstalled dependencies, sequence)."""
-        return (self.priority, self.sequence)
+        """The key is the tuple (# uninstalled dependencies, randomizer, sequence)."""
+        return (self.priority, self.randomizer, self.sequence)
```

Comment:
```suggestion
        return (self.priority, self.attempts + self.random)
```

There's no need to special-case `self.priority == 0`, personally I find ^ easier to follow.

I don't think `self.sequence` is relevant anymore, it would tie-break two tasks deterministically in the unlikely case their random numbers are equal, there's no point in doing that deterministically. So, `_counter` and `self.sequence = next(_counter)` elsewhere can be dropped.

Reply:
For a reason that escapes me ATM, it was important in the original design that build tasks with the same priority retain the order in which they were added.  My changes here were intended to reflect that goal.

Perhaps @tgamblin @becker33 @alalazo @scheibelp remember why that was done.

Reply:
That will continue to be the case if you order by `(priority, attemps, random)` since the random number is constant.